File: Assignments/Assignment_1/Assignment_1/wordcount.scala
 

import org.apache.spark.rdd.RDD
import scala.io.Codec.string2codec
import scala.io.Source
import scala.reflect.io.File
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._
import org.apache.spark.SparkConf

object WordCount {
    def main(args: Array[String]) {
  
        /* configure spark application */
        val conf = new SparkConf().setAppName("Spark Scala WordCount Example").setMaster("local[1]")
  
        /* spark context*/
        val sc = new SparkContext(conf)
  
        /* map */
        var map = sc.textFile("text.txt").flatMap(line => line.split(" ")).map(word => (word,1))
  
        /* reduce */
        var counts = map.reduceByKey(_ + _)
  
        /* print */
        counts.collect().foreach(println)
  
        /* or save the output to file */
        counts.saveAsTextFile("result.txt")
  
        sc.stop()
    }
}

File: Assignments/Assignment_1/Assignment_1/word_count.py
from pyspark import SparkContext

import os
import sys


sc = SparkContext(appName='wordCount')

input_file_path = sys.argv[1]
textRDD = sc.textFile(input_file_path)


counts = textRDD.flatMap(lambda line: line.split(' ')) \
    .map(lambda word: (word, 1)).reduceByKey(lambda a, b: a+b).collect()

with open('result.txt', 'w') as f:
    for each_word in counts:
        print(each_word)
        f.write(str(each_word)+'\n')

f.close()
